crucible-mir/src/Mir/TransTy.hs

```diff
@@ -573,33 +572,43 @@ buildArrayLit trep exps = do
                 return $ V.cons e v'
             Nothing -> mirFail "bad type in build array"
 
-buildTuple :: [MirExp s] -> MirExp s
-buildTuple xs = exp_to_assgn (xs) $ \ctx asgn ->
-    MirExp (C.StructRepr ctx) (S.app $ E.MkStruct ctx asgn)
-
-buildTupleMaybe :: M.Collection -> [M.Ty] -> [Maybe (MirExp s)] -> Either String (MirExp s)
-buildTupleMaybe col tys xs = exp_to_assgn_Maybe col tys xs $ \ctx asgn ->
-    Right (MirExp (C.StructRepr ctx) (S.app $ E.MkStruct ctx asgn))
+buildTupleM :: [M.Ty] -> [MirExp s] -> MirGenerator h s ret (MirExp s)
+buildTupleM tys xs = buildTupleMaybeM tys (map Just xs)
 
 buildTupleMaybeM :: [M.Ty] -> [Maybe (MirExp s)] -> MirGenerator h s ret (MirExp s)
 buildTupleMaybeM tys xs = do
-  col <- use $ cs . collection
-  case buildTupleMaybe col tys xs of
-    Left err -> mirFail ("buildTupleMaybe: " ++ err)
-    Right rpr -> return rpr
-
-accessAggregateMaybe :: HasCallStack => MirExp s -> Int -> MirGenerator h s ret (MirExp s)
-accessAggregateMaybe (MirExp (C.StructRepr ctx) ag) i
-  | Just (Some idx) <- Ctx.intIndex (fromIntegral i) (Ctx.size ctx) = do
-      let tpr = ctx Ctx.! idx
-      case tpr of
-        C.MaybeRepr tpr' ->
-            let mv = R.App $ E.FromJustValue tpr' (S.getStruct idx ag)
-                    (R.App $ E.StringLit "Unitialized aggregate value")
-            in return $ MirExp tpr' mv
-        _ -> mirFail "accessAggregateMaybe: non-maybe struct"
+    col <- use $ cs . collection
+    ag <- mirAggregate_uninit (fromIntegral $ length tys)
+    ag' <- foldM
+        (\ag (i, ty, mExp) -> do
+            case mExp of
+                Just (MirExp tpr rv) -> mirAggregate_set i 1 tpr rv ag
+                Nothing -> return ag)
+        ag (zip3 [0..] tys xs)
+    return $ MirExp MirAggregateRepr ag'
+
+getTupleElem :: HasCallStack => [M.Ty] -> MirExp s -> Int -> MirGenerator h s ret (MirExp s)
+getTupleElem tys tupleExp i = do
+    ty <- case tys ^? ix i of
+        Just x -> return x
+        Nothing -> mirFail $ "getTupleElem: index " ++ show i ++ " out of range for " ++ show tys
+    getTupleElemTyped tupleExp i ty
 
-accessAggregateMaybe (MirExp ty a) b = mirFail $ "invalid access of " ++ show ty ++ " at field (maybe) " ++ (show b)
+getTupleElemTyped :: HasCallStack => MirExp s -> Int -> M.Ty -> MirGenerator h s ret (MirExp s)
+getTupleElemTyped (MirExp tpr ag) i ty = do
+    col <- use $ cs . collection
+    case isZeroSized col ty of
+        False -> do
+            let tpr' = MirAggregateRepr
+            Refl <- testEqualityOrFail tpr tpr' $
+                "getTupleElem: expected tuple to be " ++ show tpr' ++ ", but got " ++ show tpr
```

Comment:
s/getTupleElem/getTupleElemTyped/

Reply:
Fixed in 1b2f3f5d2057ff19b64a058133bfe55ae4bc0c69